### PR TITLE
replace github.com/pkg/errors with braces.dev/errtrace

### PR DESCRIPTION
**Pull request by coryb**
mostly mechanical change with a few manual corrections

### CODE REVIEW

File: examples/build/main.go [hunk 1]
```diff
@@ -9,6 +9,7 @@ import (
 	"os"
 	"runtime"
 
+	"braces.dev/errtrace"
```
Comment (aaronlehmann) [inline examples/build/main.go]: Not sure we need `errtrace` in an example, but I don't object to it either.

File: frontend.go [hunk 2]
```diff
@@ -99,7 +99,7 @@ func Frontend(source string, opts ...FrontendOption) llb.State {
 
 		sess := LoadSession(ctx)
 		if sess == nil {
-			return llb.Scratch(), errors.Errorf("frontend solve request without active session")
+			return llb.Scratch(), errtrace.Errorf("frontend solve request without active session")
```
Comment (aaronlehmann) [inline frontend.go]:
```suggestion
			return llb.Scratch(), errtrace.New("frontend solve request without active session")
```

File: sockproxy/sockprovider.go [hunk 3]
```diff
@@ -36,10 +36,10 @@ func NewProvider(confs []AgentConfig) (session.Attachable, error) {
 	for _, conf := range confs {
 		if !conf.SSH {
 			if len(conf.Paths) != 1 {
-				return nil, errors.Errorf("must provide a path to a socket")
+				return nil, errtrace.Errorf("must provide a path to a socket")
```
Comment (aaronlehmann) [inline sockproxy/sockprovider.go]:
```suggestion
				return nil, errtrace.New("must provide a path to a socket")
```

File: sockproxy/sockprovider.go [hunk 4]
```diff
@@ -36,10 +36,10 @@ func NewProvider(confs []AgentConfig) (session.Attachable, error) {
...
 			}
 			if _, ok := m[conf.ID]; ok {
-				return nil, errors.Errorf("invalid duplicate ID %s", conf.ID)
+				return nil, errtrace.Errorf("invalid duplicate ID %s", conf.ID)
```
Comment (aaronlehmann) [inline sockproxy/sockprovider.go]:
```suggestion
				return nil, errtrace.Errorf("invalid duplicate ID %q", conf.ID)
```

File: sockproxy/sockprovider.go [hunk 5]
```diff
@@ -50,18 +50,18 @@ func NewProvider(confs []AgentConfig) (session.Attachable, error) {
 		}
 
 		if conf.Paths[0] == "" {
-			return nil, errors.Errorf("invalid empty ssh agent socket, make sure SSH_AUTH_SOCK is set")
+			return nil, errtrace.Errorf("invalid empty ssh agent socket, make sure SSH_AUTH_SOCK is set")
```
Comment (aaronlehmann) [inline sockproxy/sockprovider.go]:
```suggestion
			return nil, errtrace.New("invalid empty ssh agent socket, make sure SSH_AUTH_SOCK is set")
```

File: sockproxy/sockprovider.go [hunk 6]
```diff
@@ -50,18 +50,18 @@ func NewProvider(confs []AgentConfig) (session.Attachable, error) {
...
 		if err != nil {
-			return nil, err
+			return nil, errtrace.Wrap(err)
 		}
 		if conf.ID == "" {
 			conf.ID = sshforward.DefaultID
 		}
 		if _, ok := m[conf.ID]; ok {
-			return nil, errors.Errorf("invalid duplicate ID %s", conf.ID)
+			return nil, errtrace.Errorf("invalid duplicate ID %s", conf.ID)
```
Comment (aaronlehmann) [inline sockproxy/sockprovider.go]:
```suggestion
			return nil, errtrace.Errorf("invalid duplicate ID %q", conf.ID)
```

File: sockproxy/sockprovider.go [hunk 7]
```diff
@@ -89,7 +89,7 @@ func (sp *socketProvider) CheckAgent(ctx context.Context, req *sshforward.CheckA
 		id = req.ID
 	}
 	if _, ok := sp.m[id]; !ok {
-		return &sshforward.CheckAgentResponse{}, errors.Errorf("unset ssh forward key %s", id)
+		return &sshforward.CheckAgentResponse{}, errtrace.Errorf("unset ssh forward key %s", id)
```
Comment (aaronlehmann) [inline sockproxy/sockprovider.go]:
```suggestion
		return &sshforward.CheckAgentResponse{}, errtrace.Errorf("unset ssh forward key %q", id)
```

File: sockproxy/sockprovider.go [hunk 8]
```diff
@@ -105,7 +105,7 @@ func (sp *socketProvider) ForwardAgent(stream sshforward.SSH_ForwardAgentServer)
 
 	src, ok := sp.m[id]
 	if !ok {
-		return errors.Errorf("unset ssh forward key %s", id)
+		return errtrace.Errorf("unset ssh forward key %s", id)
```
Comment (aaronlehmann) [inline sockproxy/sockprovider.go]:
```suggestion
		return errtrace.Errorf("unset ssh forward key %q", id)
```

File: sockproxy/sockprovider.go [hunk 9]
```diff
@@ -154,35 +154,35 @@ func toAgentSource(paths []string) (source, error) {
...
 		}
 		fi, err := os.Stat(p)
 		if err != nil {
-			return source{}, errors.WithStack(err)
+			return source{}, errtrace.Wrap(err)
 		}
 		if fi.Mode()&os.ModeSocket > 0 {
 			if keys {
-				return source{}, errors.Errorf("invalid combination of keys and sockets")
+				return source{}, errtrace.Errorf("invalid combination of keys and sockets")
```
Comment (aaronlehmann) [inline sockproxy/sockprovider.go]:
```suggestion
				return source{}, errtrace.New("invalid combination of keys and sockets")
```

File: sockproxy/sockprovider.go [hunk 10]
```diff
@@ -154,35 +154,35 @@ func toAgentSource(paths []string) (source, error) {
...
 			}
 			socket = p
 			continue
 		}
 		keys = true
 		f, err := os.Open(p)
 		if err != nil {
-			return source{}, errors.Wrapf(err, "failed to open %s", p)
+			return source{}, errtrace.Errorf("failed to open %s: %w", p, err)
```
Comment (aaronlehmann) [inline sockproxy/sockprovider.go]:
```suggestion
			return source{}, errtrace.Errorf("failed to open %q: %w", p, err)
```

File: sockproxy/sockprovider.go [hunk 11]
```diff
@@ -115,14 +115,14 @@ func (sp *socketProvider) ForwardAgent(stream sshforward.SSH_ForwardAgentServer)
 		var err error
 		conn, err = net.DialTimeout("unix", src.socket, time.Second)
 		if err != nil {
-			return errors.Wrapf(err, "failed to connect to %s", src.socket)
+			return errtrace.Errorf("failed to connect to %s: %w", src.socket, err)
```
Comment (aaronlehmann) [inline sockproxy/sockprovider.go]:
```suggestion
			return errtrace.Errorf("failed to connect to %q: %w", src.socket, err)
```

File: sockproxy/sockprovider.go [hunk 12]
```diff
@@ -154,35 +154,35 @@ func toAgentSource(paths []string) (source, error) {
...
 		keys = true
 		f, err := os.Open(p)
 		if err != nil {
-			return source{}, errors.Wrapf(err, "failed to open %s", p)
+			return source{}, errtrace.Errorf("failed to open %s: %w", p, err)
 		}
 		dt, err := io.ReadAll(&io.LimitedReader{R: f, N: 100 * 1024})
 		if err != nil {
-			return source{}, errors.Wrapf(err, "failed to read %s", p)
+			return source{}, errtrace.Errorf("failed to read %s: %w", p, err)
```
Comment (aaronlehmann) [inline sockproxy/sockprovider.go]:
```suggestion
			return source{}, errtrace.Errorf("failed to read %q: %w", p, err)
```

File: sockproxy/sockprovider.go [hunk 13]
```diff
@@ -154,35 +154,35 @@ func toAgentSource(paths []string) (source, error) {
...
 		}
 
 		k, err := ssh.ParseRawPrivateKey(dt)
 		if err != nil {
-			return source{}, errors.Wrapf(err, "failed to parse %s", p) // TODO: prompt passphrase?
+			return source{}, errtrace.Errorf("failed to parse %s: %w", p, err) // TODO: prompt passphrase?
```
Comment (aaronlehmann) [inline sockproxy/sockprovider.go]:
```suggestion
			return source{}, errtrace.Errorf("failed to parse %q: %w", p, err) // TODO: prompt passphrase?
```

File: sockproxy/sockprovider.go [hunk 14]
```diff
@@ -154,35 +154,35 @@ func toAgentSource(paths []string) (source, error) {
...
 		}
 		if err := a.Add(agent.AddedKey{PrivateKey: k}); err != nil {
-			return source{}, errors.Wrapf(err, "failed to add %s to agent", p)
+			return source{}, errtrace.Errorf("failed to add %s to agent: %w", p, err)
```
Comment (aaronlehmann) [inline sockproxy/sockprovider.go]:
```suggestion
			return source{}, errtrace.Errorf("failed to add %q to agent: %w", p, err)
```

File: sockproxy/sockprovider.go [hunk 15]
```diff
@@ -210,21 +210,21 @@ type readOnlyAgent struct {
 }
 
 func (a *readOnlyAgent) Add(_ agent.AddedKey) error {
-	return errors.Errorf("adding new keys not allowed by buildkit")
+	return errtrace.Errorf("adding new keys not allowed by buildkit")
```
Comment (aaronlehmann) [inline sockproxy/sockprovider.go]:
```suggestion
	return errtrace.New("adding new keys not allowed by buildkit")
```

File: sockproxy/sockprovider.go [hunk 16]
```diff
@@ -210,21 +210,21 @@ type readOnlyAgent struct {
...
 }
 
 func (a *readOnlyAgent) Remove(_ ssh.PublicKey) error {
-	return errors.Errorf("removing keys not allowed by buildkit")
+	return errtrace.Errorf("removing keys not allowed by buildkit")
```
Comment (aaronlehmann) [inline sockproxy/sockprovider.go]:
```suggestion
	return errtrace.New("removing keys not allowed by buildkit")
```

File: sockproxy/sockprovider.go [hunk 17]
```diff
@@ -210,21 +210,21 @@ type readOnlyAgent struct {
...
 }
 
 func (a *readOnlyAgent) RemoveAll() error {
-	return errors.Errorf("removing keys not allowed by buildkit")
+	return errtrace.Errorf("removing keys not allowed by buildkit")
```
Comment (aaronlehmann) [inline sockproxy/sockprovider.go]:
```suggestion
	return errtrace.New("removing keys not allowed by buildkit")
```

File: sockproxy/sockprovider.go [hunk 18]
```diff
@@ -210,21 +210,21 @@ type readOnlyAgent struct {
...
 }
 
 func (a *readOnlyAgent) Lock(_ []byte) error {
-	return errors.Errorf("locking agent not allowed by buildkit")
+	return errtrace.Errorf("locking agent not allowed by buildkit")
```
Comment (aaronlehmann) [inline sockproxy/sockprovider.go]:
```suggestion
	return errtrace.New("locking agent not allowed by buildkit")
```

File: sockproxy/sockprovider.go [hunk 19]
```diff
@@ -210,21 +210,21 @@ type readOnlyAgent struct {
...
 }
 
 func (a *readOnlyAgent) Extension(_ string, _ []byte) ([]byte, error) {
-	return nil, errors.Errorf("extensions not allowed by buildkit")
+	return nil, errtrace.Errorf("extensions not allowed by buildkit")
```
Comment (aaronlehmann) [inline sockproxy/sockprovider.go]:
```suggestion
	return nil, errtrace.New("extensions not allowed by buildkit")
```

File: yaml.go [hunk 20]
```diff
@@ -472,12 +472,12 @@ func (g graphState) yamlMount(op *pb.Op, m *pb.Mount) (*yaml.Node, error) {
...
 	}
 	if int(m.Input) >= len(op.Inputs) {
-		return nil, errors.Errorf("invalid op, impossible input %d from op with %d inputs", m.Input, len(op.Inputs))
+		return nil, errtrace.Errorf("invalid op, impossible input %d from op with %d inputs", m.Input, len(op.Inputs))
 	}
 
 	input, err := g.visit(op.Inputs[m.Input])
 	if err != nil {
-		return nil, errors.Wrapf(err, "visiting mount %s", m.Dest)
+		return nil, errtrace.Errorf("visiting mount %s: %w", m.Dest, err)
```
Comment (aaronlehmann) [inline yaml.go]:
```suggestion
		return nil, errtrace.Errorf("visiting mount %q: %w", m.Dest, err)
```

File: yaml.go [hunk 21]
```diff
@@ -620,19 +620,19 @@ func (g graphState) yamlBuildOp(op *pb.Op, b *pb.BuildOp) (*yaml.Node, error) {
 		} else {
 			input, err := g.visit(op.Inputs[b.Inputs[name].Input])
 			if err != nil {
-				return nil, errors.Wrapf(err, "visiting buildOp input %s", name)
+				return nil, errtrace.Errorf("visiting buildOp input %s: %w", name, err)
```
Comment (aaronlehmann) [inline yaml.go]:
```suggestion
				return nil, errtrace.Errorf("visiting buildOp input %q: %w", name, err)
```

**Review by aaronlehmann — APPROVED** [review]

File: examples/build/main.go (hunk 1 above)
Comment (coryb) [reply examples/build/main.go]: that was automatically added by their tool, agree it is not necessary for example.

**Pull request merged**